README.md

```diff
@@ -35,11 +35,22 @@ add_arc(digraph, a, b, 4.005);
 add_arc(digraph, a, c, 253);
```

Comment:
Might also be useful to know how to create a digraph from scratch, how to add a new node or edge... specific usage of the digraph struct. 

For example, what exactly does the above call to add_arc do? Are the arguments strictly typed? If so, what are the types? a and b could be any variable (I know that they are nodes, but a reader might not), 253 could be an int/float/double/etc. Be more specific about the inputs and outputs of each useful function.

This should include all constructors; how to create a digraph and what exactly can be done with it once it is made. (What you have for the adjacency matrix stuff and shortest paths is perfect, but things like add_node or add_arc also need to be documented).